File: src/main/java/io/github/dylmeadows/eontimer/model/resource/FxmlResource.java
package io.github.dylmeadows.eontimer.model.resource;

import io.github.dylmeadows.common.core.io.Resource;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
public enum FxmlResource implements Resource {
    Gen3TimerPane("io/github/dylmeadows/eontimer/fxml/timer/Gen3TimerPane.fxml"),
    Gen4TimerPane("io/github/dylmeadows/eontimer/fxml/timer/Gen4TimerPane.fxml"),
    Gen5TimerPane("io/github/dylmeadows/eontimer/fxml/timer/Gen5TimerPane.fxml"),
    TimerDisplayPane("io/github/dylmeadows/eontimer/fxml/TimerDisplayPane.fxml"),
    TimerMasterPane("io/github/dylmeadows/eontimer/fxml/TimerMasterPane.fxml");

    private final String path;

}

File: src/main/kotlin/io/github/dylmeadows/eontimer/model/resource/FxmlResource.kt
package io.github.dylmeadows.eontimer.model.resource

import io.github.dylmeadows.common.core.io.Resource

enum class FxmlResource(private val path: String) : Resource {
    EonTimerPane("EonTimerPane.fxml"),
    Gen3TimerPane("timer/Gen3TimerPane.fxml"),
    Gen4TimerPane("timer/Gen4TimerPane.fxml"),
    Gen5TimerPane("timer/Gen5TimerPane.fxml"),
    CustomTimerPane("timer/CustomTimerPane.fxml"),
    TimerControlPane("timer/TimerControlPane.fxml"),
    ActionSettingsPane("settings/ActionSettingsPane.fxml"),
    TimerSettingsPane("settings/TimerSettingsPane.fxml"),
    SettingsControlPane("settings/SettingsControlPane.fxml"),
    TimerDisplayPane("TimerDisplayPane.fxml");

    override fun getPath(): String {
        return "$BASE_RESOURCE_PATH/fxml/${this.path}"
    }
}